g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/source/dao/impl/S3SourceDaoImpl.java

```diff
@@ -0,0 +1,87 @@
+//Copyright 2019-2020 VMware, Inc.
+//SPDX-License-Identifier: EPL-2.0
+package com.vmware.l10n.source.dao.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+
+import com.vmware.l10n.record.dao.SqlLiteDao;
+import com.vmware.l10n.source.dao.SourceDao;
+import com.vmware.l10n.utils.S3Util;
+import com.vmware.l10n.utils.SourceUtils;
+import com.vmware.vip.common.i18n.dto.SingleComponentDTO;
+import com.vmware.vip.common.l10n.source.dto.ComponentMessagesDTO;
+
+@Repository
+@Profile("s3")
+public class S3SourceDaoImpl implements SourceDao {
+
+	private static Logger logger = LoggerFactory.getLogger(S3SourceDaoImpl.class);
+
+	@Autowired
+	private S3Util s3util;
+
+	@Autowired
+	private SqlLiteDao sqlLite;
+
+	/**
+	 * the path of local resource file,can be configured in spring config file
+	 **/
+	@Value("${source.bundle.file.basepath}")
+	private String basePath;
+
+	@Override
+	public String getFromBundle(SingleComponentDTO componentMessagesDTO) {
+		logger.info("Read content from file: {}/{}", componentMessagesDTO.getLocale(),
+				componentMessagesDTO.getComponent());
+
+		return s3util.readBundle(basePath, componentMessagesDTO);
+	}
+
+	@Override
+	public boolean updateToBundle(ComponentMessagesDTO compDTO) {
+		logger.info("[Save sources to storage]: {}/{}/{}/{}", compDTO.getProductName(), compDTO.getVersion(),
+				compDTO.getComponent(), compDTO.getLocale());
+
+		boolean bExist = false;
+		try {
```

Comment:
1. 
  Here is already the entrance of DAO layer. If throw the exception, it will propagate to service layer.
2. 
  How to add a mark for Logz? Do you mean add a log message when failure?

Reply:
1. We can throw it upper than controller, you can refer to the manager project on how to handle it.
2. Please check with Devops side, i think we just add a mark as part of the error message so that the Logz can recognize it.

Reply:
I'm afraid that it needs to change the overall flow and is out of scope of this PR. This PR only focus on the DAO layer.

Reply:
done